virtool/uploads/db.py

```diff
@@ -0,0 +1,38 @@
+from typing import Union
+
+from sqlalchemy.ext.asyncio import AsyncSession
+
+import virtool.utils
+from virtool.uploads.models import Upload
+
+
+async def create(db, name: str, upload_type: str, reserved: bool = False, user: Union[None, str] = None):
+    """
+    Creates and commits a new upload document
+
+    :param db: PostgreSQL database object
+    :param name: the name of the upload
+    :param upload_type: the type of upload (e.g. reads)
+    :param reserved: should the file immediately be reserved (used for legacy samples)
+    :param user: the id of the uploading user
+    :return: a JSON response document
+    """
+    async with AsyncSession(db) as session:
+        upload = Upload(
+            created_at=virtool.utils.timestamp(),
+            name=name,
+            ready=False,
+            removed=False,
+            reserved=reserved,
+            type=upload_type,
+            user=user
+        )
+
+        session.add(upload)
+
+        await session.flush()
+        document = upload.to_dict()
+
+        await session.commit()
+
+        return document
```

Comment:
Simplify to:
```py
await session.commit()

return upload.to_dict()
```

Reply:
I was wanting to have it like this but I seem to lose access to `upload` completely after the commit.

This is what `upload` refers to before the `commit()`:
`<Upload(id=None, created_at=2015-10-06 20:00:00, name=Test.fq.gz, name_on_disk=None, ready=False, removed=False, reserved=False, size=None, type=hmm, user=test, uploaded_at=None>`

After:
`Unable to get repr for <class 'virtool.uploads.models.Upload'>`